app/models/page.rb

```diff
@@ -117,13 +115,19 @@ def obsolete?
   end
 
   def redirect_page
-    if redirect_to.present?
-      @redirect_page ||= self.class.find_by_path(redirect_to, base: base, try_index: false).filename
+    lifecycle = parsed_file["lifecycle"]
+    return nil if lifecycle.nil?
+
+    if lifecycle["redirect_to"].present?
+      @redirect_page ||= self.class.find_by_path(lifecycle["redirect_to"], base: base, try_index: false).filename
     end
   end
 
   def expired?
-    expires_at.present? && expires_at.past?
```

Comment:
[suggestion] I think we should revert this change, since we have `expires_at` available through the changes to `has_field` This version just duplicates the `through` logic that exists there.

Reply:
I will give this a shot